lib/std/mem.zig

```diff
@@ -234,6 +238,7 @@ pub const Allocator = struct {
         const bytes_len = bytes.len + @boolToInt(Slice.sentinel != null);
```

Comment:
I noticed this additional byte for sentinel strings in here. Is that correct? And why don't I see similar code in the other functions?

Reply:
Everywhere else conveniently gets to operate based on the "length", which does not include the bytes of the sentinel element. However, here, we are interested in the actual number of bytes required in memory, which includes the sentinel, if any.

Looking at this, I do see a bug: it should be `@boolToInt(Slice.sentinel != null) * @sizeOf(Slice.child)`. The number in master branch is incorrect for sentinel-terminated pointers with element types wider than 1 byte.

Consider, when allocating a null-terminated slice, one must ask for + 1 element, set it to null, and then do a sentinel-terminated slice syntax. The resulting type is what will be passed to free(), so you can see it has to do the + 1 to match